File: exercise/sample/perl/parsing_html.pl
#!/usr/bin/perl

use strict;
use warnings;

=head1 DESCRIPTION

This example script retrieves the html page from uniprot and extract
information from it.

dependencies:
perl-HTML-Tree

=cut

# print ASCII from HTML from a URL
use LWP::Simple;
use HTML::Parse;
my ($html, $ascii);
$html = get("http://www.uniprot.org/uniprot/Q42435");
defined $html
    or die "Can't fetch HTML from uniprot";

# Retrieve the protein name
$html =~ m/class="FULL">([\D\s]+?)<\/span>/;
my $name = $1;
print "Protein has for name: $name \n";

File: exercise/sample/python/parsing_html.py
#!/usr/bin/python
#-*- coding: utf-8 -*-

"""
Parse uniprot's html to extract some information for the protein.
"""

import re
import requests

if __name__ == '__main__':
    # Load the information from uniprot
    html = requests.get('http://www.uniprot.org/uniprot/Q42435')

    # Retrieve the name of the protein
    m = re.search('class="FULL">([\D\s]+?)</span>', html.content)
    name = m.group(1)
    print 'Protein has for name         : %s' % name
